frontend/packages/knative-plugin/src/components/add/channels/ChannelForm.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues, useFormikContext } from 'formik';
+import { FormFooter, FlexForm, useFormikValidationFix } from '@console/shared';
+import { LoadingInline } from '@console/internal/components/utils';
+import {
+  isDefaultChannel,
+  getChannelKind,
+  getChannelData,
+  useDefaultChannelConfiguration,
+} from '../../../utils/create-channel-utils';
+import { ChannelListProps } from '../import-types';
+import YamlSection from './sections/YamlSection';
+import FormViewSection from './sections/FormViewSection';
+import ChannelSelector from './form-fields/ChannelSelector';
+
+interface OwnProps {
+  namespace: string;
+  channels: ChannelListProps;
+}
+
+const ChannelForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
+  errors,
+  handleSubmit,
+  handleReset,
+  status,
+  isSubmitting,
+  dirty,
+  namespace,
+  channels,
+}) => {
+  const {
+    values,
+    setFieldValue,
+    setFieldTouched,
+    validateForm,
+    setErrors,
+    setStatus,
+  } = useFormikContext<FormikValues>();
+  useFormikValidationFix(values);
+  const [defaultChannel, defaultChannelLoaded] = useDefaultChannelConfiguration(namespace);
+  const channelHasFormView = values.type && isDefaultChannel(getChannelKind(values.type));
+  const channelKind = getChannelKind(values.type);
+  const onTypeChange = React.useCallback(
+    (item: string) => {
+      setErrors({});
+      setStatus({});
+      const kind = getChannelKind(item);
+      if (isDefaultChannel(kind) && kind.length > 0) {
+        const nameData = `data.${kind.toLowerCase()}`;
+        const sourceData = getChannelData(kind.toLowerCase());
+        setFieldValue(nameData, sourceData);
+        setFieldTouched(nameData, true);
+      }
+      setFieldValue('yamlData', '');
+      setFieldTouched('yamlData', true);
+
+      setFieldValue('type', item);
+      setFieldTouched('type', true);
+
+      setFieldValue('name', _.kebabCase(`${kind}`));
+      setFieldTouched('name', true);
+      validateForm();
+    },
+    [setErrors, setStatus, setFieldValue, setFieldTouched, validateForm],
+  );
+
+  return (
+    <FlexForm onSubmit={handleSubmit}>
+      {((channels && !channels.loaded) || !defaultChannelLoaded) && <LoadingInline />}
```

Comment:
we should show loading only if `channels?.loaded` is false and not depend on `defaultChannelLoaded` as even if channel is loaded and `defaultConfiguredChannel` couldn't be identified we can still show list of channels WDYT?

Reply:
I added the check, reason being If we are not accounting the wait time of deafult Channel Discovery, then we will be seeing sudden change in the form view. There are use cases where, if default channel does not have form view, in that scenario the UI will suddenly switch from form view to yaml view. To avoid those flickering and to provide smooth look and feel i have added it.